### ISSUE TITLE
`gcp-sql-database-require-ssl` requires setting removed `require_ssl` property

### ISSUE THREAD

**Issue by sugar700** [returntocorp/sgrep-rules#3505]
**Describe the bug**
The `terraform.gcp.security.gcp-sql-database-require-ssl.gcp-sql-database-require-ssl` requires setting `require_ssl`. The problem with that is Google Terraform Provider 6.0 removed this flag: https://registry.terraform.io/providers/hashicorp/google/latest/docs/guides/version_6_upgrade#settingsip_configurationrequire_ssl-is-now-removed-in-601

**To Reproduce**
```terraform
resource "google_sql_database_instance" "main" {
  name      = "some-example-name"
  database_version  = "POSTGRES_15"
  region      = "europe-west3"
  settings {
    tier = "db-f1-micro"
    ip_configuration {
      ssl_mode = "TRUSTED_CLIENT_CERTIFICATE_REQUIRED"
    }
  }
}
```

**Expected behavior**
Semgrep should allow setting `ssl_mode` to `TRUSTED_CLIENT_CERTIFICATE_REQUIRED`.

**Priority**
How important is this to you?
- [ ] P0: blocking me from making progress
- [ ] P1: this will block me in the near future
- [x] P2: annoying but not blocking me

**Additional Context**
